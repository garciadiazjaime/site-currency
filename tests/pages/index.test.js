import React from 'react'
import { shallow } from 'enzyme'

import IndexPage from '../../pages/index'

jest.mock('../../utils/rateUtil', () => ({
  renderRatesFor() {
    return 'happy-rates'
  }
}));

describe('IndexPage', () => {
  describe('main', () => {
    const wrapper = shallow(<IndexPage />)

    it('matches snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('calls renderRatesFor', () => {
      expect(wrapper.text().includes('happy-rates')).toBe(true);
    })
  })
})
