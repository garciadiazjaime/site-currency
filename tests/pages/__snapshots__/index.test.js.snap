// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage main matches snapshot 1`] = `
<section
  className="jsx-1578069544"
>
  <SideEffect(Head)>
    <title
      className="jsx-1578069544"
    >
      Valor del Peso Mexicano en el Mundo
    </title>
    <meta
      className="jsx-1578069544"
      content="Cu치ntos pesos mexicanos necesito para comprar un d칩lar y otras monedas"
      name="description"
    />
  </SideEffect(Head)>
  <h1
    className="jsx-1578069544"
  >
    <img
      alt=""
      className="jsx-1578069544"
      src="/static/images/flags/MXN.jpg"
    />
    Peso Mexicano
  </h1>
  <h2
    className="jsx-1578069544"
  >
    Cu치ntos pesos necesito para un:
  </h2>
  <RatesList />
  <h2
    className="jsx-1578069544"
  >
    Cu치ntos pesos necesito para un:
  </h2>
  <JSXStyle
    css="h1.jsx-1578069544{background:#000;color:#FFF;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin:0px;padding:12px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}h2.jsx-1578069544{margin:12px 12px 0 12px;font-size:18px;font-weight:normal;}img.jsx-1578069544{width:40px;height:40px;margin-right:10px;}"
    styleId="1578069544"
  />
</section>
`;
